clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1427,7 +1427,7 @@ private void acquire() {
         ensureNotClosed();
         long threadId = Thread.currentThread().getId();
         if (threadId != currentThread.get() && !currentThread.compareAndSet(NO_CURRENT_THREAD, threadId))
-            throw new ConcurrentModificationException("KafkaConsumer is not safe for multi-threaded access");
+            throw new ConcurrentModificationException("KafkaConsumer is not safe for multi-threaded access. Request accessing thread is " +threadId + " and it is already being accessed by "+currentThread.get());
```

Comment:
Basic patch looks fine, but this code formatting won't pass checkstyle. After cleaning up you can verify it passes with `./gradlew clients:checkstyleMain`.